token/client/src/token.rs

```diff
@@ -536,26 +537,28 @@ where
     }
 
     /// Assign a new authority to the account.
-    pub async fn set_authority<S: Signer>(
+    pub async fn set_authority<S: Signers>(
         &self,
         account: &Pubkey,
+        authority: &Pubkey,
         new_authority: Option<&Pubkey>,
         authority_type: instruction::AuthorityType,
-        owner: &S,
-    ) -> TokenResult<()> {
+        signing_keypairs: &S,
+    ) -> TokenResult<T::Output> {
         self.process_ixs(
             &[instruction::set_authority(
                 &self.program_id,
                 account,
                 new_authority,
                 authority_type,
-                &owner.pubkey(),
+                authority,
+                // XXX i think this is wrong and we need to supply all bulk signers for multisig to work
+                // leaving for now, theres a working example in close_account
                 &[],
```

Comment:
im 90% sure this would cause multisig shit to fail because of `validate_owners` in `program-2022/src/processor.rs`, theres a (in theory???) working but ugly example below to compare

Reply:
You got it right -- this definitely doesn't work with multisigs unfortunately.